### ISSUE TITLE
Strip block containing filters lost all variables and filter variables from php doc

### ISSUE THREAD

**Issue by lulco** [efabrica-team/phpstan-latte#191]
Found thanks to https://github.com/spaze/michalspacek.cz/issues/51

Generated code looks like this:

```php
    \ob_start(fn() => '');
    try {
        (function () {
            /* line 2 */
            echo \Latte\Runtime\Filters::escapeHtmlText(\strlen($title));
            echo "\n";
            /* line 3 */
            echo \Latte\Runtime\Filters::escapeHtmlText(\strlen(1, 23, 45));
            echo "\n";
            /* line 4 */
            echo \Latte\Runtime\Filters::escapeHtmlText($__filter__translate->translate($title));
        })(\get_defined_vars());
    } finally {
        $ʟ_fi = new \Latte\Runtime\FilterInfo('html');
        echo \Latte\Runtime\Filters::convertTo($ʟ_fi, 'html', $this->filters->filterContent('strip', $ʟ_fi, \ob_get_clean()));
    }
```

problem is  `function () {}` it has no use, in original code there is extract of params, we removed it. Also fn can be used but only in new php versions and only with single expression

**Comment by lulco** [comment]
In older version of latte compiler there was just `try {...} finally {...}` and I think we should change it back here. So remove function () part and keep just inner code

**Comment by MartinMystikJonas** [comment]
I think better solution would be to rewrite AddVarTypesVisitor in a way that it is added everywhere where `extract($this->params)` is used

Remove of function can hide relevant errors (like use of local variable from outer context inside function body) 

Why we remove and re-add extracts anyway? Was there some conflict? Maybe remove it only in top-level of method?

**Comment by lulco** [comment]
That was first implementation, but extract isn't there always. We changed it to begin of each method few weeks ago. 

Also this is just testing the nette / latte itself, not template. It worked without function() before, it will now, too.

**Comment by MartinMystikJonas** [comment]
I think this behaviour is one of BC breaks Latte 3 did. So we should not ignore errors it causes. 

Add types to every method is correct. I just do not know why we need to remove and re-add extract.

**Comment by lulco** [comment]
It's long story :)

First, I've added php doc for vars after exract(this->params), but if there are no params sent to template, no extract is present (so filter variables are not defined). You suggest to add php docs at the start of each method, so I did it. But then the extract was after all these phpdocs and phpstan throws errors about non-existing variables. So I remove the original extract and add one to the start of methods (after phpdocs were added). Everything works.

Anyway, I'm pretty sure we can remove this closure in latte 3 compiled templates and move its statements directly to try catch block. It will have no effect for latte tests. We will also remove another error which it caused - "calling closure with one parameter when 0 is required."
